homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -97,7 +148,16 @@ def update(self):
                 _LOGGER.debug("No response from inverter (could be dark)")
             else:
                 raise error
-            self._state = None
         finally:
+            self._available = bool(self._state is not None)
+            if self._available != self.availableprev:
```

Comment:
Why store previous available as an instance variable? Just store it locally inside the update method before you start updating.

Reply:
Done